### PR TITLE
test(ticdc): migrate test-infra to testify for pkg/util pkg

### LINKED ISSUE #4879: migrate test-infra to testify for `pkg/util` pkg

### PR DESCRIPTION
**Pull request by mikechengwei**
### What problem does this PR solve?
Issue Number: close #4879

### What is changed and how it works?

This pr migrate test-infra to testify for pkg/util pkg .

Tests 

 - Unit test

Code changes

Side effects

Related changes

### Release note 

```release-note
None .
```

### CODE REVIEW

File: pkg/util/bitflag_test.go [hunk 1]
```diff
@@ -25,12 +27,8 @@ const (
...
 )
 
-type bitFlagSuite struct{}
-
-var _ = check.Suite(&bitFlagSuite{})
-
-func (b *bitFlagSuite) TestExample(c *check.C) {
-	defer testleak.AfterTest(c)()
+func TestExample(t *testing.T) {
+	defer testleak.NewAfterTest(t)()
```
Comment (ben1009) [inline pkg/util/bitflag_test.go:31]: i think we do not need `testleak` after migration to testify, try https://github.com/pingcap/tiflow/blob/master/pkg/leakutil/leak_helper.go

File: pkg/util/testleak/fake.go [hunk 2]
```diff
@@ -31,6 +31,12 @@ func AfterTest(c *check.C) func() {
 	}
 }
 
+// NewAfterTest is a new dummy implementation by testify when build tag 'leak' is not set.
+func NewAfterTest(c *testing.T) func() {
+	return func() {
+	}
+}
+
```
Comment (overvenus) [inline pkg/util/testleak/fake.go:39]: Do we still need it?

Comment (mikechengwei) [reply pkg/util/testleak/fake.go]: Thks for remind.

**Review by ben1009 — COMMENTED** [review]
does the pkg have leaktest enabled ?
pls update the pr description accordingly `This pr migrate test-infra to testify for pkg/util pkg and add testleak.NewAfterTest method.`

**Review by ben1009 — APPROVED** [review]

**Review by hi-rustin — APPROVED** [review]

**Comment by hi-rustin** [comment]
Thanks for working on this. Next time you can `/cc @hi-rustin`. I will be notified to review your PR.

**Comment by CharlesCheung96** [comment]
> does the pkg have leaktest enabled ? pls update the pr description accordingly `This pr migrate test-infra to testify for pkg/util pkg and add testleak.NewAfterTest method.`

@mikechengwei We need to add a `main_test` file to enable testleak. Ref: https://github.com/pingcap/tiflow/pull/3103/files#diff-dd6dc527bf649b2345d44e2632a68f638b59565e5a215b285ca0417615b8a4c3R23

**Comment by hi-rustin** [comment]
/hold

https://github.com/pingcap/tiflow/pull/5036#issuecomment-1085343261

**Review by hi-rustin — APPROVED** [review]

**Pull request merged**